doc/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,67 @@
+# Troubleshooting Production Issues related to Unsupported Operations in Mountpoint
+
+Mountpoint enables users to access S3 buckets through local filesystem. While this allows for existing workflows and applications to work with S3 unchanged, it also limits the effective API for Mountpoint to be a subset of the filesystem API.
+So, there are a few operations in filesystem API which are not supported using Mountpoint on a mounted bucket. Here, we are enumerating those common cases with their errors reported in the logs which customers face. Please look at [SEMANTICS document](doc/SEMANTICS.md) for more information on what operations are allowed in Mountpoint.
+
+## Non-sequential Write
+Mountpoint only supports sequential writes. On random out of order writes users should expect `Invalid Argument`. For example - 
+```
+$ dd if=/dev/random of=out seek=1 count=1
+dd: writing to 'out': Invalid argument
+```
+In the logs, we get the following WARN message - 
+```
+WARN write{req=52 ino=49 fh=3 offset=512 length=512 name="out"}: mountpoint_s3::fuse: write failed:\
+upload error: out of order write NOT supported by Mountpoint, aborting the upload; expected offset 0 but got 512
+```
+
+## Writing to an existing File 
+Mountpoint does support overwriting to existing file using mount option `--allow-overwrite` . Trying to open an existing file for writing without `--allow-overwrite`  flag will fail with the error: `Operation not permitted`.
+In the logs, we get the following WARN message - 
+```
+WARN file overwrite is disabled by default, you need to remount with --allow-overwrite flag and open the file in truncate mode (O_TRUNC) to overwrite it
+```
+
+## Deleting file (`rm` without mount option `--allow-delete`)
+In order to delete files using Mountpoint user need to enable it using `--allow-delete` mount option. To know more about the behaviour of file deletion, please visit [Delete Semantics](https://github.com/awslabs/mountpoint-s3/blob/main/doc/SEMANTICS.md#deletes)
+Without the flag, if a customer tries to run  `rm test-file.txt` command, they will see the following error: `rm: cannot remove 'test-file.txt': Operation not permitted`
+In the logs, this will show up as 
+```
+WARN unlink{req=8 parent=1 name="test-file.txt"}: mountpoint_s3::fuse: unlink failed: Deletes are disabled. \
+Use '--allow-delete' mount option to enable it.
+```
+
+## Listing file which is shadowed by directory of same name
+A file system directory cannot contain both a file and a directory of the same name. If your bucket's directory structure would result in this state, only the directory will be accessible. For example, our bucket contains the following object keys:
```

Comment:
Is this true? What happens with S3 Express One Zone?

Reply:
Yes. It is true for normal bucket. But, for S3 Express One zone, we are able to see both the file and directory. Also, this warning is not emitted. 
But, still reading the file is not possible. It gives error - 
`cat: out: Is a directory`